cirq-rigetti/cirq_rigetti/service.py

```diff
@@ -104,14 +96,11 @@ def sampler(self) -> RigettiQCSSampler:
         )
 
     @staticmethod
-    @_provide_default_client
-    def list_quantum_processors(
-        client: Optional[httpx.Client],
-    ) -> ListQuantumProcessorsResponse:  # pragma: no cover
+    def list_quantum_processors(client: Optional[QCSClient]) -> List[str]:  # pragma: no cover
```

Comment:
I think this needs the default `client = None` so that the client is truly optional.

Also instead of `no cover`, please consider adding a unit test with `unittest.mock`-ed client object.

Reply:
Changed the signature to have a default `None`, added a unit test, and removed `no cover`